sh/sh.go

```diff
@@ -43,8 +44,47 @@ func (r *Runner) apply(opts ...RunnerOption) {
 	}
 }
 
+type taskWriter struct {
+	w          io.Writer
+	taskName   string
+	streamName string
+}
+
+// taskWriter prefixes each line of the output with a timestamp and the task name.
+func (t taskWriter) Write(p []byte) (n int, err error) {
+	msgLines := bytes.Split(p, []byte{'\n'})
+
+	// if original message ends with a newline, remove the last empty line
+	if len(msgLines) > 0 && len(msgLines[len(msgLines)-1]) == 0 {
+		msgLines = msgLines[:len(msgLines)-1]
+	}
```

Comment:
There's a nice function for this in the standard library: [TrimRight](https://pkg.go.dev/bytes#TrimRight). :slightly_smiling_face: 
```suggestion
	msgLines := bytes.Split(bytes.TrimRight(p, "\n"), []byte{'\n'})
```